texar/modules/encoders/xlnet_encoders.py

```diff
@@ -0,0 +1,602 @@
+            if lr is None:
+                raise ValueError(
+                    "lr must be specified when lr_layer_decay_rate is not 1.0")
+
+            num_layers = self._hparams.num_layers
+            scope = self.variable_scope.name
+            base_var_names = ['r_w_bias', 'r_r_bias', 'word_embedder']
+
+            if self._hparams.use_segments:
+                base_var_names.extend(['r_s_bias', 'seg_embed'])
+
+            base_params = [tf.trainable_variables(scope=scope + "/" + name)[0]
+                           for name in base_var_names]
+            base_group = {
+                "params": base_params,
+                "lr": lr * (lr_layer_scale ** num_layers
+                            if decay_base_params else 1.0)
+            }
+            param_groups = [base_group]
+            for idx in range(num_layers):
+                decay_rate = lr_layer_scale ** (num_layers - idx - 1)
+                param_group = {
+                    "params": tf.trainable_variables(
+                        scope=scope + "/" + "layer_{}".format(idx)),
+                    "lr": lr * decay_rate,
+                }
+                param_groups.append(param_group)
+        else:
+            param_groups = self.trainable_variables
+        return param_groups
+
+    @property
+    def output_size(self):
+        r"""Return the output size of the network.
+        """
+        return self._hparams.hidden_dim
+
+    @staticmethod
+    def _cache_mem(curr_out, prev_mem, mem_len, reuse_len=None):
+        r"""Cache hidden states into memory."""
+        assert mem_len > 0
+
+        if reuse_len is not None and reuse_len > 0:
+            curr_out = curr_out[:reuse_len]
+
+        if prev_mem is None:
+            new_mem = curr_out[-mem_len:]
+        else:
+            new_mem = tf.concat([prev_mem, curr_out], 0)[-mem_len:]
+
+        return tf.stop_gradient(new_mem)
+
+    def _create_mask(self, qlen, mlen, dtype=tf.float32, same_length=False):
+        r"""Create causal attention mask."""
+        attn_mask = tf.ones([qlen, qlen], dtype=dtype)
+        mask_u = tf.matrix_band_part(attn_mask, 0, -1)
+        mask_dia = tf.matrix_band_part(attn_mask, 0, 0)
+        attn_mask_pad = tf.zeros([qlen, mlen], dtype=dtype)
+        ret = tf.concat([attn_mask_pad, mask_u - mask_dia], axis=1)
+        if same_length:
+            mask_l = tf.matrix_band_part(attn_mask, -1, 0)
+            ret = tf.concat([ret[:, :qlen] + mask_l - mask_dia, ret[:, qlen:]],
+                            axis=1)
+
+        return ret
+
+    def _build(self, token_ids, segment_ids=None, input_mask=None,
+               memory=None, permute_mask=None, target_mapping=None,
+               bi_data=False, clamp_len=None, cache_len=0, same_length=False,
+               attn_type='bi', two_stream=False, mode=None):
+        r"""Compute XLNet representations for the input.
+
+        Args:
+            token_ids: Shape `[batch_size, seq_len]`.
```

Comment:
In other modules' doc we say `[batch_size, max_time]`, instead of `seq_len`. https://github.com/asyml/texar/blob/master/texar/modules/decoders/rnn_decoders.py#L303

Pls keep the terminology consistent. Also, if PyTorch has the same issue, pls fix too

Reply:
Fixed in 056c949. Issue in texar-pytorch https://github.com/asyml/texar-pytorch/issues/144

Reply:
Fixed in 056c949. Issue in texar-pytorch https://github.com/asyml/texar-pytorch/issues/144